plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -526,6 +527,57 @@ public function get_all_order_column_mappings() {
 		return $this->all_order_column_mapping;
 	}
 
+	/**
+	 * The group name to use when caching order object data.
+	 *
+	 * @return string
+	 */
+	private function get_cache_group(): string {
+		return 'orders_data';
+	}
+
+	/**
+	 * Delete cached order data for the given object_ids.
+	 *
+	 * @internal This method should only be used by internally and in cases where the CRUD operations of this datastore
+	 *           are bypassed for performance purposes. This interface is not guaranteed.
+	 *
+	 * @param array $object_ids The IDs of the orders to remove cache.
+	 *
+	 * @return bool[] Array of return values, grouped by the object_id. Each value is either true on success, or false
+	 *                if the contents were not deleted.
+	 */
+	public function invalidate_cache_for_objects( array $object_ids ): array {
```

Comment:
The dependency it has on the internals of the datastore class are how objects are cached. Caching is meant to be opaque, so what happens in this method is specific to this implementation, e.g, what cache group/keying to use or if we add a similar implementation on top of a CPT datastore, we would likely call `clean_post_cache()` within this method. 

Ideally, this method wouldn't even need to be public since the fact that there is caching isn't really something that should be known outside of the datastore.  Ideally, we also wouldn't have to call the meta datastore to invalidate its cache for an object. But, because there are existing pathways that directly modify data that the `OrdersTableDataStore` manages, we need a way for those paths to be able to notify the datastore that the data was modified externally.

Would it make more sense to leave this method here if the method were called something like `::objects_modified_externally( array $object-ids )`?